### PR TITLE
refactor(relayer): Simplify deposit confirmation config

### PR DESCRIPTION
**Pull request by pxrl**
DEFAULT_MIN_DEPOSIT_CONFIRMATIONS isn't actually used anymore, so drop it, but migrate the chain finality defaults into
MIN_DEPOSIT_CONFIRMATIONS. Also, split out testnet chain IDs to a specific tier. This should make it a bit simpler to update and maintain in future.

While here, drop Goerli MDC configs since they're no longer needed. This change also includes the pending Mode deposit confirmation config from PR #1582.

### CODE REVIEW

**Comment by pxrl** [comment]
Background: This was also needed to fix the relayer on testnet.

File: src/common/Constants.ts [hunk 1]
```diff
@@ -32,79 +32,57 @@ export const FINALIZER_TOKENBRIDGE_LOOKBACK = 14 * 24 * 60 * 60;
...
+    8453: 120,
+    34443: 120,
+    42161: 0,
+    59144: 30,
+  },
   1000: {
     1: 32, // Justified block
     10: 60,
     137: 100, // Probabilistically safe level based on historic Polygon reorgs
-    288: 0,
```
Comment (nicholaspai) [inline src/common/Constants.ts:77]: So what will happen to 280 now?

File: src/relayer/RelayerConfig.ts [hunk 2]
```diff
@@ -268,12 +268,15 @@ export class RelayerConfig extends CommonConfig {
...
     // Append default thresholds as a safe upper-bound.
-    Object.keys(this.minDepositConfirmations).forEach((chainId) =>
-      this.minDepositConfirmations[chainId].push({
-        usdThreshold: bnUint256Max,
-        minConfirmations: Number.MAX_SAFE_INTEGER,
-      })
-    );
+    Object.keys(this.minDepositConfirmations).forEach((chainId) => {
+      const depositConfirmations = this.minDepositConfirmations[chainId];
+      if (depositConfirmations.at(-1).minConfirmations < Number.MAX_SAFE_INTEGER) {
```
Comment (nicholaspai) [inline src/relayer/RelayerConfig.ts:273]: Never seen this at(-1) syntax before. Is it the same as the "last" element?

File: src/common/Constants.ts (hunk 1 above)
Comment (pxrl) [reply src/common/Constants.ts:77]: Nothing - this is zkSync Görli and this PR is removing definitions for the Görli chains.

File: src/relayer/RelayerConfig.ts (hunk 2 above)
Comment (pxrl) [reply src/relayer/RelayerConfig.ts:273]: Yep - it's a shorter version of `depositConfirmations[depositConfirmations.length - 1]`.

File: src/common/Constants.ts (hunk 1 above)
Comment (nicholaspai) [reply src/common/Constants.ts:77]: sorry i meant 288

Comment (pxrl) [reply src/common/Constants.ts:77]: Hm, yeah it's also nuked on this. Do we think Boba will ever come back? It's easy enough to re-add it if it ever gets enabled again, but until that point it's not required.

File: src/relayer/RelayerConfig.ts (hunk 2 above)
Comment (james-a-morris) [reply src/relayer/RelayerConfig.ts:273]: As a note - `.at(-1)` does return `Value | undefined` so if we ever flip to check undefined then we will error on this

**Review by james-a-morris — COMMENTED** [review]
One question https://github.com/across-protocol/relayer-v3/pull/1546#discussion_r1613926446

**Review by nicholaspai — APPROVED** [review]

**Pull request merged**